lib/post_creator.rb

```diff
@@ -159,7 +159,7 @@ def valid?
         return false
       end
 
-      unless @topic.present? && (@opts[:skip_guardian] || guardian.can_create?(Post, @topic))
+      unless @topic.present? && (@opts[:skip_guardian] || guardian.can_create?(Post, @topic) || guardian.can_perform_action_available_to_group_moderators?(@topic))
```

Comment:
Have you considered amending `can_create_topic` to include the moderator logic? That way the logic stays in the guardian and not the post creator service object.